lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -0,0 +1,241 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class Shift4Gateway < Gateway
+      self.test_url = 'https://utgapi.shift4test.com/api/rest/v1/'
+      self.live_url = 'https://utg.shift4api.net/api/rest/v1/'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = %i[visa master american_express discover]
+
+      self.homepage_url = 'https://utg.shift4api.net'
+      self.display_name = 'Shift4'
+
+      STANDARD_ERROR_CODE_MAPPING = {}
+
+      def initialize(options = {})
+        requires!(options, :client_guid, :auth_token)
+        super
+      end
+
+      def access_token(options = {})
+        post = {}
+        add_credentials(post, options)
+        commit('accesstoken', post, request_headers(options))
+      end
+
+      def purchase(money, options = {})
```

Comment:
Method definitions for transactions need to have a payment_method in the signature. See examples here: https://github.com/activemerchant/active_merchant/wiki/Contributing#generating-the-gateway-skeleton

Reply:
Further information on payment methods: https://github.com/activemerchant/active_merchant/wiki/Patterns-&-Standards#payment_method

Reply:
Done